samcli/lib/test_runner/generate_env_vars.py

```diff
@@ -0,0 +1,94 @@
+import yaml
+from typing import List
+from samcli.commands.exceptions import FailedArnParseException
+
+
+class FargateRunnerArnMapGenerator:
+    def _get_env_var_name(self, resource_arn: str) -> str:
```

Comment:
Do we want any checks to ensure we don't collide with reserved env variables? I'm thinking of aws related ones that the fargate task needs to have to support auth, our own variables for S3 bucket, requirements, etc. 

It may be unlikely, but I am worried customers might break in a confusing way if they have an unfortunately named resource.

Reply:
This is done in `run` process: https://github.com/aws/aws-sam-cli/blob/d81eec8ddbb9c1ede36cb14947f51c0c76c13314/samcli/lib/test_runner/invoke_testsuite.py#L79-L90

It is also impossible to have any collisions by default, since all generated environment variable names begin with an underscore, and none of the reserved names do. The `init` output warns of the reserved names so that the customer is aware of which names they cannot add or change existing ones to.